includes/Core/Authentication/Google_Proxy.php

```
@@ -95,6 +95,72 @@ public function get_site_fields() {
 		);
 	}
 
+	/**
+	 * Fetch site fields
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param Credentials $credentials Credentials instance.
+	 * @return array|WP_Error The response as an associative array or WP_Error on failure.
+	 */
+	public function fetch_site_fields( Credentials $credentials ) {
+		if ( ! $credentials->has() ) {
+			return null;
```

@benbowler let's return an error if we don't have credentials.